### ISSUE TITLE
`drain_fifo` seems redundant vs `read_bytes`

### ISSUE THREAD

**Issue by MabezDev** [esp-rs/esp-hal#2725]

**Comment by bugadani** [comment]
What is `fill_fifo`? I can't find it in `uart.rs` at all.

**Comment by MabezDev** [comment]
> What is `fill_fifo`? I can't find it in `uart.rs` at all.

Sorry, it's `drain_fifo`

**Comment by bugadani** [comment]
`read_bytes` waits for the buffer to fill up, but I agree we probably don't need two of these.

**Comment by bjoernQ** [comment]
> `read_bytes` waits for the buffer to fill up, but I agree we probably don't need two of these. We should probably go with the non-blocking semantics of dran_fifo and let the user handle the waiting.

I agree with the exception that `drain_fifo` is a weird name in public API - something like `try_read`? (or maybe not - I'm bad at naming but I can spot weird naming when I see it 😆 )
